nexus_constructor/model/load_from_json.py

```diff
@@ -53,9 +78,217 @@ def _retrieve_children_list(json_dict: dict) -> list:
         return []
 
 
+def _create_transformation_dataset(
+    angle_or_magnitude: float, dtype: str, name: str
+) -> Dataset:
+    """
+    Creates the transformation dataset using the angle/magnitude, name, and dtype of the transformation.
+    :param angle_or_magnitude: The angle or magnitude.
+    :param dtype: The data type.
+    :param name: The transformation name.
+    :return: A dataset containing the above information.
+    """
+    return Dataset(
+        name, dataset=DatasetMetadata(size=[1], type=dtype), values=angle_or_magnitude,
+    )
+
+
+class TransformationReader:
+    def __init__(self, parent_component: Component, entry: list):
+        self.parent_component = parent_component
+        self.entry = entry
+        self.warnings = []
+
+    def add_transformations_to_component(self):
+        """
+        Attempts to construct Transformation objects using information from the JSON dictionary and then add them to the
+        parent component.
+        """
+        for item in self.entry:
+            if _contains_transformations(item):
+                try:
+                    self._create_transformations(item["children"])
+                except KeyError:
+                    continue
+
+    def _get_transformation_property(
+        self,
+        property_name: str,
+        json_transformation: dict,
+        transform_name: str = None,
+        failure_value: Any = None,
+        parse_result_func: Callable = None,
```

Comment:
I can't spot a benefit to passing `parse_result_func` into this method. Why not simply call it on the result?